.travis.yml

```diff
@@ -42,32 +40,61 @@ install:
-
-matrix:
-  fast_finish: true
+jobs:
   include:
-    - python: "3.4"
-      env: PART="--exclude-dir=theano/tensor -e test_flake8.py ." THEANO_FLAGS="mode=FAST_COMPILE"
-    - python: "3.4"
-      env: PART="theano/tensor --exclude-dir=theano/tensor/nnet" THEANO_FLAGS="mode=FAST_COMPILE"
-    - python: "3.4"
-      env: PART="theano/tensor/nnet" THEANO_FLAGS="mode=FAST_COMPILE"
-    - python: "2.7"
-      env: PART="--exclude-dir=theano/tensor -e test_flake8.py ." THEANO_FLAGS="mode=FAST_COMPILE,floatX=float32"
-    - python: "2.7"
-      env: PART="theano/tensor --exclude-dir=theano/tensor/nnet" THEANO_FLAGS="mode=FAST_COMPILE,floatX=float32"
-    - python: "2.7"
-      env: PART="theano/tensor/nnet" THEANO_FLAGS="mode=FAST_COMPILE,floatX=float32"
+    # define prototype for doctest
+    - &doctest
+      stage: doc
+      python: "2.7"
+      env: DOC=1 PART="theano/tests/test_flake8.py"
+    # re-use prototype, changing the Python version
+    - <<: *doctest
+      python: "3.4"
+    - &normaltest
+      stage: test
+      python: "2.7"
+      env: PART="theano/compat theano/compile theano/d3viz theano/gof theano/misc theano/sandbox theano/scalar theano/scan_module theano/sparse theano/tests -e test_flake8.py theano/typed_list"
+    - <<: *normaltest
+      python: "3.4"
+    - <<: *normaltest
+      env: PART="theano/tensor -e test_basic.py --exclude-dir=theano/tensor/nnet"
+    - <<: *normaltest
+      python: "3.4"
+      env: PART="theano/tensor -e test_basic.py --exclude-dir=theano/tensor/nnet"
+    - <<: *normaltest
+      env: PART="theano/tensor/tests/test_basic.py"
+    - <<: *normaltest
+      python: "3.4"
+      env: PART="theano/tensor/tests/test_basic.py"
+    - <<: *normaltest
+      env: PART="theano/tensor/nnet -e test_abstract_conv.py"
+    - <<: *normaltest
+      python: "3.4"
+      env: PART="theano/tensor/nnet -e test_abstract_conv.py"
+    - <<: *normaltest
+      env: PART="theano/tensor/nnet/tests/test_abstract_conv.py"
+    - <<: *normaltest
+      python: "3.4"
+      env: PART="theano/tensor/nnet/tests/test_abstract_conv.py"
+    - <<: *normaltest
+      env: FAST_COMPILE=1 PART="--exclude-dir=theano/tensor -e test_flake8.py ."
+    - <<: *normaltest
+      python: "3.4"
+      env: FAST_COMPILE=1 PART="--exclude-dir=theano/tensor -e test_flake8.py ."
+    - <<: *normaltest
+      env: FAST_COMPILE=1 PART="theano/tensor --exclude-dir=theano/tensor/nnet"
+    - <<: *normaltest
+      python: "3.4"
+      env: FAST_COMPILE=1 PART="theano/tensor --exclude-dir=theano/tensor/nnet"
+    - <<: *normaltest
+      env: FAST_COMPILE=1 PART="theano/tensor/nnet"
+    - <<: *normaltest
+      python: "3.4"
+      env: FAST_COMPILE=1 PART="theano/tensor/nnet"
 
 script:
+  - if [[ $FAST_COMPILE == "1" ]]; then export THEANO_FLAGS=$THEANO_FLAGS,mode=FAST_COMPILE; fi
+  - if [[ $FAST_COMPILE == "1" && $TRAVIS_PYTHON_VERSION == '2.7' ]]; then export THEANO_FLAGS=$THEANO_FLAGS,floatX=float32; fi
```

Comment:
I don't like this. Why not set directly the THEANO_FLAGS in the "env:" line? Yes, you prevent to duplicate that flag, but it make the code more complicated and hide the config difference about floatX=float32. Can you remove line 97 and set floatX=float32 in the "env:" line? Or make an FLOATX env variable and do the same equivalent to FAST_COMPILE env var? That would not hide that flag change.